encrypted-media-respec.html

```diff
@@ -1732,13 +1732,53 @@ <h2><dfn>MediaKeySession</dfn> Interface</h2>
       <p>For methods that return a promise, all errors are reported asynchronously by rejecting the returned Promise. This includes [[WEBIDL]] type mapping errors.</p>
       <p>The following steps of an algorithm are always aborted when rejecting a promise.</p>
 
+<div><pre class="idl">enum MediaKeySessionClosedReason {
+    "internal-error",
+    "closed-by-application",
+    "release-acknowledged",
+    "hardware-context-reset",
+    "resource-evicted"
+};</pre>
+<p>
+The <dfn>MediaKeySessionClosedReason</dfn> enumeration is defined as follows:
+</p>
+<table class="simple" data-dfn-for="MediaKeySessionClosedReason" data-link-for="MediaKeySessionClosedReason"><tbody><tr><th colspan="2">Enumeration description</th></tr><tr><td><dfn><code id="idl-def-MediaKeySessionClosedReason.unknown">internal-error</code></dfn></td><td>
+            The session was closed because of an unrecoverable error in the CDM.
+            <div class="note">
+              <p>
+                When this occurs, applications SHOULD NOT expect to be able to create new sessions on this <a>MediaKeys</a> instance.
```

Comment:
Drive-by nit: Don't put normative "SHOULD NOT" language inside notes. https://respec.org/docs/#conformance-section claims that Respec flags this, but I don't see an error in your preview.

You could flip the wording to "This MediaKeys instance might not be able to host new sessions." or similar, or elevate the whole thing to normative as "Applications SHOULD NOT attempt to create ...". I wouldn't use "SHOULD NOT expect", as that's making normative requirements on an author's mental state, which is weird. :)

Reply:
Thank you for this!  The language of spec is still new for me.